### ISSUE TITLE
Updated conversation interface

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#122]
```
I would like to have the option of allowing replies made by the player to be 
longer than what we currently have. Currently, we are limited to the amount of 
text that fits onto a standard Android button, and we usually have replies such 
as "Yes", "No", "I have it" and so on, so that the text fits.

Once we start developing the quest, you will have to choose sides and justify 
your actions to other NPCs that you find, and this will require player replies 
that are far longer than what we can do now.

I would also like to have the ability for the player to see previous phrases 
within a certain conversation, so you can review what the NPC said.

I'm thinking we redesign the conversation interface to something similar to 
what is found in the Bioware games (for example Baldur's Gate), where the 
conversation is a scrollable text view, and every player reply can be radio 
buttons.
```

Original issue reported on code.google.com by `oskar.wi...@gmail.com` on 9 Jan 2011 at 8:43

**Comment by GoogleCodeExporter** [comment]
```
Definitely. When in conversation, the whole screen could be taken up and the 
character who is talking would have their color text one color or with their 
name in front of each reply. Maybe even both characters player icon on each 
side. Then the player replies could cover that up and disappear as replies were 
chosen. 
```

Original comment by `Captain8...@gmail.com` on 9 Jan 2011 at 6:10

**Comment by GoogleCodeExporter** [comment]
```
I have made a first try at a new conversation interface. See screenshots below. 

The conversation log is scrollable, so you can see previous statements from 
both actors. When selecting a reply and pressing "Next", the radio buttons are 
replaced with the reply you selected.

Any thoughts? Should we move the radio buttons somewhere else? Should we keep 
the "Leave" button?

With this way of doing conversations, it feels more natural to have a 
conversation rather than the NPC monolog that we had before I think. This way 
we could also in the future have several NPCs talking in the same conversation.
```

Original comment by `oskar.wi...@gmail.com` on 13 Jan 2011 at 4:27
- Changed state: **Started**

Attachments:
- [conversation1.png](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-122/comment-2/conversation1.png)
- [conversation2.png](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-122/comment-2/conversation2.png)

**Comment by GoogleCodeExporter** [comment]
```
Looks great. This opens up gameplay further down the road where you can choose 
different options similar to the alternate endings with the Vacor/Unzel 
situation. It allows players to give their character personality. Although that 
may be a little more than you had in mind. It looks good though. Good size 
buttons, the whole conversation is viewable, very streamlined. How would it 
look if there were more conversation options? Like 3 buttons... 
```

Original comment by `Captain8...@gmail.com` on 13 Jan 2011 at 5:20

**Comment by GoogleCodeExporter** [comment]
```
Marvelous!
Some hints from other games:
- Wiz8, Baldur's Gate: personality influences communication formulae.
- Goldbox games: you may choose to be sly, meek, abusive, haughty...
- TOEE: your skills also influence the range of choices; e.g. successful 
intimidate, bluff or diplomacy checks. 
- Buck Rogers: Fast Talk is simply a generic skill that helps you in all 
conversations to get on the "winning" side of the conversation (be it bluff, 
intimidation or diplomacy).

Some of these need character personality -- not necessarily as a fixed value; 
more like built up from the player's choice of responses. If a player 
consistently chooses to be abusive with NPCs (or certain NPCs, e.g. women or 
thieves), that may give him a reputation later on.

Others need skills. as in the TOEE example. More aspects are spot and sense 
motive, or even wisdom / intelligence that helps character combine past 
information into new questions. This is implemented in Wiz8 I think; you have a 
list of keywords, and if your party has high enough skills for information 
exchange then they "learn" new keywords faster.

Let me give you an example.
Someone is talking about Unzel who's an apprentice mage.
- Unzel is of course added to the list of "known" things (keywords).
- If the character has average perception then "apprentice mage" will be added 
too.
- If the character has high perception then "master mage" will be added as 
well, since any apprentice must have a master.

Another example.
A character is talking to Unzel.
- If he has good spotting skill, he may ask about the campfire or even Unzel's 
chest (iirc he has a chest). Not that these mean anything, but good subjects 
for idle chat :)
```

Original comment by `surrano` on 13 Jan 2011 at 5:42

**Comment by GoogleCodeExporter** [comment]
```
Very nice! Well done. We should update the conversations.
"Bread" > "Do you need some bread?"
"What did you mean with ...?"
"You told me about: ... Can you tell me more about it."
etc.
```

Original comment by `SamuelPl...@gmail.com` on 13 Jan 2011 at 10:58

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.8 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 24 Feb 2011 at 9:33
- Changed state: **Fixed**
